addons/binding/org.openhab.binding.onkyo/src/main/java/org/openhab/binding/onkyo/internal/OnkyoAlbumArt.java

```diff
@@ -0,0 +1,188 @@
+                } else {
+                    state = State.INVALID;
+                }
+                packetFlagStr = "Start";
+                albumArtStringBuilder.setLength(0);
+                break;
+            case '1':
+                packetFlagStr = "Next";
+                if (state == State.STARTED || state == State.NEXT) {
+                    state = State.NEXT;
+                } else {
+                    state = State.INVALID;
+                }
+                break;
+            case '2':
+                packetFlagStr = "End";
+                if (state == State.STARTED || state == State.NEXT) {
+                    state = State.READY;
+                } else {
+                    state = State.INVALID;
+                }
+                break;
+            case '-':
+                packetFlagStr = "notUsed";
+                state = State.READY;
+                break;
+            default:
+                state = State.INVALID;
+                logger.debug("Unknown album art packet flag '{}'", packetFlag);
+        }
+
+        if (state != State.INVALID) {
+            switch (imageType) {
+                case BMP:
+                case JPEG:
+                    String picData = data.substring(2, data.length());
+                    logger.debug("Received album art fragment in '{}' format, packet flag '{}', picData '{}'",
+                            imageType, packetFlagStr, picData);
+                    albumArtStringBuilder.append(picData);
+                    break;
+                case URL:
+                    coverArtUrl = data.substring(2);
+                    logger.debug("Received album art url '{}'", coverArtUrl);
+                    break;
+                case NONE:
+                    logger.debug("Received information: album art not available");
+                    break;
+                default:
+            }
+
+        } else {
+            logger.debug("Received album art fragment in wrong order, format '{}', packet flag '{}'", imageType,
+                    packetFlagStr);
+        }
+    }
+
+    public byte[] getAlbumArt() throws IllegalArgumentException {
+        byte[] data = null;
+
+        if (state == State.READY) {
+            switch (imageType) {
+                case BMP:
+                case JPEG:
+                    data = DatatypeConverter.parseHexBinary(albumArtStringBuilder.toString());
+                    break;
+                case URL:
+                    data = downloadAlbumArt(coverArtUrl);
+                    break;
+                case NONE:
+                default:
+            }
+
+            return data;
+        }
+
+        throw new IllegalArgumentException("Illegal Album Art");
+    }
+
+    private byte[] downloadAlbumArt(String coverArtUrl) {
+        try {
+            URL url = new URL(coverArtUrl);
+            URLConnection connection = url.openConnection();
+            return IOUtils.toByteArray(connection.getInputStream());
```

Comment:
Is it guaranteed that the input stream is closed by IOUtils? Afaik not, so you might rather want to use try-with-resources here.